runtime/doc/lua-plugin.txt

```diff
@@ -0,0 +1,450 @@
+        strategy = { cfg.strategy, "string" },
+    })
+end
+<
+And invalid config will result in an error message like 
+`"vim.g.my_plugin.strategy: expected string, got number"`.
+
+By doing this, you can use the validation both
+
+- At initialization, reporting errors with |vim.notify()|.
+- In a |health| check.
+
+==============================================================================
+Troubleshooting					   *lua-plugin-troubleshooting*
+
+------------------------------------------------------------------------------
+Health checks			       	    *lua-plugin-troubleshooting-health*
+
+Provide health checks in `lua/{plugin}/health.lua`.
+
+Some things to validate:
+
+- User cnfiguration
+- Proper initializaion
+- Presence of Lua dependencies (e.g. other plugins)
+- Presence of external dependencies
+
+See also |vim.health and |health-dev|.
+
+------------------------------------------------------------------------------
+Minimal config template             *lua-plugin-troubleshooting-minimal-config*
+
+It can be useful to provide a template for a minimal configuration, along with 
+a guide on how to use it to reproduce issues.
+
+==============================================================================
+Versioning and releases                        *lua-plugin-versioning-releases*
+
+Consider
+
+- Using SemVer https://semver.org/ tags and releases to properly communicate
+  bug fixes, new features, and breaking changes.
+- Automating versioning and releases in CI.
+- Publishing to luarocks https://luarocks.org, especially if your plugin
+  has dependencies or components that need to be built; or if it could be a
+  dependency for another plugin.
+
+------------------------------------------------------------------------------
+Further reading		       *lua-plugin-versioning-releases-further-reading*
+
+- Luarocks <3 Neovim https://github.com/nvim-neorocks/sample-luarocks-plugin
+
+------------------------------------------------------------------------------
+Tools			                 *lua-plugin-versioning-releases-tools*
+
+- luarocks-tag-release
+  https://github.com/marketplace/actions/luarocks-tag-release
+- release-please-action
+  https://github.com/marketplace/actions/release-please-action
+- semantic-release 
+  https://github.com/semantic-release/semantic-release 
+
+==============================================================================
+Documentation                                        *lua-plugin-documentation*
+
+Provide vimdoc (see |help-writing|), so that users can read your plugin's
+documentation in Neovim, by entering `:h {plugin}` in |command-mode|.
+
+------------------------------------------------------------------------------
+Tools			                       *lua-plugin-documentation-tools*
+
+- panvimdoc https://github.com/kdheepak/panvimdoc
+- lemmy-help https://github.com/numToStr/lemmy-help
+
+NOTE: lemmy-help has not been updated recently and is no longer fully
+      compatible with |LuaCATS|.
```

Comment:
Not sure if we want to add this here.
It hasn't been touched in about 2 years. I still use it and it works fine (as long as I don't use newer features like `(exact)` classes, but I'm either going to fork it or switch to something else.

Reply:
LMAO absolutely remove this what

Reply:
If it's deprecated and not recommended then people don't need to know. Just yeet it.